### PR TITLE
NIFI-5245: SimpleCSVFileLookupService should take account of charset

### PR DESCRIPTION
**Pull request by ssogabe**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by MikeThomsen** [comment]
You have two checkstyle issues that caused the build to fail. I'll start a review, but those need to be fixed.

File: nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestCSVRecordLookupService.java [hunk 1]
```diff
@@ -66,4 +66,30 @@ public void testSimpleCsvFileLookupService() throws InitializationException, IOE
...
+
+        runner.addControllerService("csv-file-lookup-service", service);
+        runner.setProperty(service, CSVRecordLookupService.CSV_FILE, "src/test/resources/test_Windows-31J.csv");
+        runner.setProperty(service, CSVRecordLookupService.CSV_FORMAT, "RFC4180");
+        runner.setProperty(service, CSVRecordLookupService.CHARSET, "Windows-31J");
+        runner.setProperty(service, CSVRecordLookupService.LOOKUP_KEY_COLUMN, "key");
+        runner.enableControllerService(service);
+        runner.assertValid(service);
+
+        final CSVRecordLookupService lookupService =
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestCSVRecordLookupService.java]: This isn't necessary.

File: nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestCSVRecordLookupService.java [hunk 2]
```diff
@@ -66,4 +66,30 @@ public void testSimpleCsvFileLookupService() throws InitializationException, IOE
...
+                (CSVRecordLookupService) runner.getProcessContext()
+                        .getControllerServiceLookup()
+                        .getControllerService("csv-file-lookup-service");
+
+        assertThat(lookupService, instanceOf(LookupService.class));
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestCSVRecordLookupService.java]: Neither is this. If you can successfully add it to the test runner and enable it, you're good to go on this.

File: nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestSimpleCsvFileLookupService.java [hunk 3]
```diff
@@ -64,4 +64,28 @@ public void testSimpleCsvFileLookupService() throws InitializationException, IOE
...
+        runner.addControllerService("csv-file-lookup-service", service);
+        runner.setProperty(service, SimpleCsvFileLookupService.CSV_FILE, "src/test/resources/test_Windows-31J.csv");
+        runner.setProperty(service, SimpleCsvFileLookupService.CSV_FORMAT, "RFC4180");
+        runner.setProperty(service, SimpleCsvFileLookupService.CHARSET, "Windows-31J");
+        runner.setProperty(service, SimpleCsvFileLookupService.LOOKUP_KEY_COLUMN, "key");
+        runner.setProperty(service, SimpleCsvFileLookupService.LOOKUP_VALUE_COLUMN, "value");
+        runner.enableControllerService(service);
+        runner.assertValid(service);
+
+        final SimpleCsvFileLookupService lookupService =
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestSimpleCsvFileLookupService.java]: These blocks can be removed.

File: nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestCSVRecordLookupService.java [hunk 4]
```diff
@@ -66,4 +66,30 @@ public void testSimpleCsvFileLookupService() throws InitializationException, IOE
...
+
+        final Optional<Record> property1 = lookupService.lookup(Collections.singletonMap("key", "property.1"));
+        assertEquals("this is property \uff11", property1.get().getAsString("value"));
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestCSVRecordLookupService.java]: I would recommend adding `assertTrue(property1.isPresent())`.

File: nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestSimpleCsvFileLookupService.java [hunk 5]
```diff
@@ -64,4 +64,28 @@ public void testSimpleCsvFileLookupService() throws InitializationException, IOE
...
+                (SimpleCsvFileLookupService) runner.getProcessContext()
+                        .getControllerServiceLookup()
+                        .getControllerService("csv-file-lookup-service");
+
+        assertThat(lookupService, instanceOf(LookupService.class));
+
+        final Optional<String> property1 = lookupService.lookup(Collections.singletonMap("key", "property.1"));
+        assertEquals(Optional.of("this is property \uff11"), property1);
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/java/org/apache/nifi/lookup/TestSimpleCsvFileLookupService.java]: Add `assertTrue(property1.isPresent())` so the test will cleanly fail the evaluation if it's not present.

**Comment by ssogabe** [comment]
Thanks @MikeThomsen, I'll fix it as soon as possible!

**Comment by MikeThomsen** [comment]
While it was building here, I ran the build locally and it passed.

**Pull request closed without merging**